spec/unit/provider/service/macosx_spec.rb

```diff
@@ -58,248 +58,288 @@
+            allow(provider).to receive(:shell_out_with_systems_locale).
+                    with(/(#{su_cmd} '#{cmd}'|#{cmd})/).
+                    and_return(double("Status",
+                                    :stdout => launchctl_stdout, :exitstatus => 0))
+            allow(provider).to receive(:shell_out_with_systems_locale!).
+                    with(/plutil -convert xml1 -o/).
+                    and_return(double("Status", :stdout => plutil_stdout))
+          end
 
-          context "when launchctl returns pid in service list" do
-            let(:launchctl_stdout) { StringIO.new <<-SVC_LIST }
-  12761 - 0x100114220.old.machinit.thing
-  7777  - io.redis.redis-server
-  - - com.lol.stopped-thing
-  SVC_LIST
-
-            before do
-              provider.load_current_resource
-            end
+          context "#{service_name} that is a #{service_type} running Osx #{platform_version}" do
+            let(:new_resource) { Chef::Resource::MacosxService.new(service_name) }
+            let!(:current_resource) { Chef::Resource::MacosxService.new(service_name) }
+
+            describe "#load_current_resource" do
+
+              # CHEF-5223 "you can't glob for a file that hasn't been converged
+              # onto the node yet."
+              context "when the plist doesn't exist" do
+
+                def run_resource_setup_for_action(action)
+                  new_resource.action(action)
+                  provider.action = action
+                  provider.load_current_resource
+                  provider.define_resource_requirements
+                  provider.process_resource_requirements
+                end
+
+                before do
+                  allow(Dir).to receive(:glob).and_return([])
+                  allow(provider).to receive(:shell_out!).
+                          with(/plutil -convert xml1 -o/).
+                          and_raise(Mixlib::ShellOut::ShellCommandFailed)
+                end
+
+                it "works for action :nothing" do
+                  expect { run_resource_setup_for_action(:nothing) }.not_to raise_error
+                end
+
+                it "works for action :start" do
+                  expect { run_resource_setup_for_action(:start) }.not_to raise_error
+                end
+
+                it "errors if action is :enable" do
+                  expect { run_resource_setup_for_action(:enable) }.to raise_error(Chef::Exceptions::Service)
+                end
+
+                it "errors if action is :disable" do
+                  expect { run_resource_setup_for_action(:disable) }.to raise_error(Chef::Exceptions::Service)
+                end
+              end
 
-            it "sets resource running state to true" do
-              expect(provider.current_resource.running).to be_truthy
-            end
+              context "when launchctl returns pid in service list" do
+                let(:launchctl_stdout) { StringIO.new <<-SVC_LIST }
```

Comment:
I agree. But I thought there was some value with being explicit with the expected value. That being said I will only define it once with this value.  As for the 3rd string, the only part that is important is the "Label" and the fact that the "PID" line is missing. If you think it make sense I can just add the important parts and exclude the rest.